FormalConjectures/Other/ReedOmegaDeltaChi.lean

```diff
@@ -0,0 +1,73 @@
+/-
+Copyright 2025 The Formal Conjectures Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-/
+
+import FormalConjectures.Util.ProblemImports
+
+/-!
+# Reed's omega, delta, and chi conjecture
+
+*References*:
+- [B. Reed,  ω Δ and χ, J. Graph Theory 27 (1998) 177-212.](https://onlinelibrary.wiley.com/doi/10.1002/(SICI)1097-0118(199804)27:4%3C177::AID-JGT1%3E3.0.CO;2-K) 
+-
+[openproblemgarden(http://www.openproblemgarden.org/op/reeds_omega_delta_and_chi_conjecture)
+- [mathoverflow/37923](https://mathoverflow.net/questions/37923) asked by user [Andrew D. King](https://mathoverflow.net/users/4580/andrew-d-king)
+-/
+
+namespace ReedOmegaDeltaChi
+
+open Classical
+
+noncomputable
+def maxDegree {V : Type} (G : SimpleGraph V) : ℕ∞ :=
+  if h : ∀ v, Finite (G.neighborSet v) then
+    have : ∀ v, Fintype (G.neighborSet v) :=
+      fun v => @Fintype.ofFinite _ (h v)
+    ⨆ v, G.degree v
+  else
+    ⊤
+
+/--
+For a nonempty graph, `G.cliqueNum` is 0 iff the clique size is unbounded. For the purposes of
+stating the conjecture, we actually want the max clique size to be `∞` in this case.
+-/
+noncomputable
+def maxCliqueSize {V : Type} (G : SimpleGraph V) : ℕ∞ :=
+  if Nonempty V ∧ G.cliqueNum = 0 then ⊤ else G.cliqueNum
+
+/--
+For a graph $G$, we define $\Delta(G)$ to be the maximum degree, $\omega(G)$ to be the size of the
+largest clique subgraph, and $\chi(G)$ to be the chromatic number. Reed's omega, delta, and chi
+conjecture states that $$\chi(G) \leq \lceil \frac{1}{2}(\omega(G) + \Delta(G) + 1) \rceil.$$
+-/
+@[category research open, AMS 5]
+theorem reed_omega_delta_chi_conjecture :
+  ∀ {V : Type} (G : SimpleGraph V),
+    let χ := G.chromaticNumber
+    let ω := maxCliqueSize G
+    let Δ := maxDegree G
+  /- (d+1)/2 = ⌈d/2⌉ -/
+  χ ≤ (ω + Δ + 1).map (fun d => (d + 1) / 2) := by
```

Comment:
Here is my suggestion, for finite graphs:
```suggestion
    (∀ {V : Type} [Fintype V] [DecidableEq V] (G : SimpleGraph V) [DecidableRel G.Adj],
      let χ := G.chromaticNumber
      let ω := G.cliqueNum
      let Δ := G.maxDegree
      /- (d+1)/2 = ⌈d/2⌉ -/
      2 * χ ≤ ω + Δ + 2 ↔ answer(sorry) := by
```

Reply:
1. Sorry, you might be right. I am an expert mathlib reviewer, but not an expert conjectures one 😅
2. But then shouldn't the infinite version be stated in terms of cardinals? Right now, you have an in-between statement.

Reply:
As I understand it, this conjecture is only for the case where the clique number and max degree are both finite, so I think we can state everything using just $N_\infty$? I'd like to defer to @andrew-d-king on this one, since I think he has a better idea than I do of the precise statement of the conjecture.

Reply:
At least, could you fix the indentation and avoid this weird `ENat.map`?
```suggestion
    (∀ {V : Type} (G : SimpleGraph V),
      let χ := G.chromaticNumber
      let ω := G.ecliqueNum
      let Δ := G.emaxDegree
      2 * χ ≤ ω + Δ + 2 ↔ answer(sorry) := by
```